src/consumer/versions-ops/checkout-version.ts

```diff
@@ -151,13 +151,13 @@ async function getComponentStatus(
       `component ${component.id.toStringWithoutVersion()} has conflicts that need to be resolved first, please use bit merge --resolve/--abort`
     );
   }
-  const getNewVersion = (): string => {
+  const getNewVersion = async (): Promise<string> => {
     // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
     if (reset) return component.id.version;
     // @ts-ignore if !reset the version is defined
-    return latestVersion ? componentModel.latest() : version;
+    return latestVersion ? componentModel.latestIncludeRemote(repo) : version;
```

Comment:
I'm not sure about this one.
let's discuss it with @davidfirst

Reply:
@GiladShoham , this returns the latest when taking into account the remote. so if locally it's 0.0.1 and the remote is 0.0.2, it'll return 0.0.2. 
It's relevant for lanes, where we save the remote head in the `.bit/refs/remote/lane-name` file.